### PR TITLE
[UI] UI Small Adjustments

### PR DESCRIPTION
**Pull request by brain-frog**
## What are the changes?

A lot of minor UI changes, will try and show them all in screenshots

## Why am I doing these changes?

Appease my brain

## What did change?

- Overlay added to menu to add contrast to menu items
- Window mask functionality updated to be easier to read (more update to come on that)
- LOTS of adding names to gameobjects, purely for debugging purposes
- Deleted deprecated images
   - command fight labels (legacy and default)
   - 678ms and msb (old way of handling shinies)
- Simplified i18n declaration module
- Moved Timed Event Display into its own class

### Screenshots/Videos

Fight Menu UI
fight-ui-update

Menu Overlay
menu-overlay

## How to test the changes?

Play locally and check for any ui that looks off

## Checklist
- [x] There is no overlap with another PR?
- [ ] The PR is self-contained and cannot be split into smaller PRs?
_Probably could be, but won't unless necessary_
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: public/images/pokemon/678msb.json
Comment (Tempo-anon) [inline public/images/pokemon/678msb.json:1]: There's only 4 images that are deprecated and removed right? I think they can be listed out on the PR desc

File: src/battle-scene.ts [hunk 1]
```diff
@@ -373,7 +375,7 @@ export default class BattleScene extends SceneBase {
     this.add.existing(transition);
 
     const uiContainer = this.add.container(0, 0);
-    uiContainer.setName("container-ui");
+    uiContainer.setName("ui");
```
Comment (Tempo-anon) [inline src/battle-scene.ts:378]: for consistency reasons should this be "uiContainer" rather than just "ui"?

File: src/battle-scene.ts [hunk 2]
```diff
@@ -413,28 +415,28 @@ export default class BattleScene extends SceneBase {
     this.fieldUI.add(this.charSprite);
 
     this.pbTray = new PokeballTray(this, true);
-    this.pbTray.setName("container-pb-tray");
+    this.pbTray.setName("pb-tray");
     this.pbTray.setup();
 
     this.pbTrayEnemy = new PokeballTray(this, false);
-    this.pbTrayEnemy.setName("container-enemy-pb-tray");
+    this.pbTrayEnemy.setName("enemy-pb-tray");
```
Comment (Tempo-anon) [inline src/battle-scene.ts:422]: name discrepancy here too "pb-tray-enemy"

File: src/battle-scene.ts [hunk 3]
```diff
@@ -813,8 +815,10 @@ export default class BattleScene extends SceneBase {
 
   addPokemonIcon(pokemon: Pokemon, x: number, y: number, originX: number = 0.5, originY: number = 0.5, ignoreOverride: boolean = false): Phaser.GameObjects.Container {
     const container = this.add.container(x, y);
+    container.setName(`${pokemon.name}-icon`);
```
Comment (Tempo-anon) [inline src/battle-scene.ts:818]: shouldn't the name include "container"?

File: src/field/damage-number-handler.ts [hunk 4]
```diff
@@ -20,6 +20,7 @@ export default class DamageNumberHandler {
     const battlerIndex = target.getBattlerIndex();
     const baseScale = target.getSpriteScale() / 6;
     const damageNumber = addTextObject(scene, target.x, -(scene.game.canvas.height / 6) + target.y - target.getSprite().height / 2, Utils.formatStat(amount, true), TextStyle.SUMMARY);
+    damageNumber.setName("text-damage-number");
```
Comment (Tempo-anon) [inline src/field/damage-number-handler.ts:23]: name discrepancy here too

File: src/ui/abstact-option-select-ui-handler.ts [hunk 5]
```diff
@@ -82,6 +84,7 @@ export default abstract class AbstractOptionSelectUiHandler extends UiHandler {
     }
 
     this.optionSelectText = addTextObject(this.scene, 0, 0, options.map(o => o.item ? `    ${o.label}` : o.label).join("\n"), TextStyle.WINDOW, { maxLines: options.length });
+    this.optionSelectText.setName("text-option-select");
```
Comment (Tempo-anon) [inline src/ui/abstact-option-select-ui-handler.ts:87]: option-select-text

File: src/ui/battle-message-ui-handler.ts [hunk 6]
```diff
@@ -31,33 +31,30 @@ export default class BattleMessageUiHandler extends MessageUiHandler {
...
+    this.bg = this.scene.add.sprite(0, 0, "bg", this.scene.windowType);
+    this.bg.setName("sprite-battle-msg-bg");
+    this.bg.setOrigin(0, 1);
+    ui.add(this.bg);
 
-    this.bg = bg;
-
-    this.commandWindow = addWindow(this.scene, 201, -1, 118, 46);
+    this.commandWindow = addWindow(this.scene, 202, 0, 118, 48);
+    this.commandWindow.setName("window-command");
```
Comment (Tempo-anon) [inline src/ui/battle-message-ui-handler.ts:40]: command-window

File: src/ui/battle-message-ui-handler.ts [hunk 7]
```diff
@@ -31,33 +31,30 @@ export default class BattleMessageUiHandler extends MessageUiHandler {
...
     this.commandWindow.setOrigin(0, 1);
     this.commandWindow.setVisible(false);
     ui.add(this.commandWindow);
 
-    this.movesWindowContainer = this.scene.add.container(1, -1);
+    this.movesWindowContainer = this.scene.add.container(0, 0);
+    this.movesWindowContainer.setName("moves-bg");
```
Comment (Tempo-anon) [inline src/ui/battle-message-ui-handler.ts:46]: moves-window-container

File: src/ui/challenges-select-ui-handler.ts [hunk 8]
```diff
@@ -41,7 +41,7 @@ export default class GameChallengesUiHandler extends UiHandler {
     const ui = this.getUi();
 
     this.challengesContainer = this.scene.add.container(1, -(this.scene.game.canvas.height / 6) + 1);
-    this.challengesContainer.setName("container-challenges");
+    this.challengesContainer.setName("challenges");
```
Comment (Tempo-anon) [inline src/ui/challenges-select-ui-handler.ts:44]: challenges-container

File: src/ui/challenges-select-ui-handler.ts [hunk 9]
```diff
@@ -116,7 +116,7 @@ export default class GameChallengesUiHandler extends UiHandler {
     this.startCursor.setVisible(false);
 
     this.valuesContainer = this.scene.add.container(0, 0);
-    this.valuesContainer.setName("container-values");
+    this.valuesContainer.setName("values");
```
Comment (Tempo-anon) [inline src/ui/challenges-select-ui-handler.ts:119]: values-container

File: src/ui/command-ui-handler.ts [hunk 10]
```diff
@@ -34,12 +34,14 @@ export default class CommandUiHandler extends UiHandler {
       i18next.t("commandUiHandler:run")
     ];
 
-    this.commandsContainer = this.scene.add.container(216, -38.7);
+    this.commandsContainer = this.scene.add.container(217, -38.7);
+    this.commandsContainer.setName("commands");
```
Comment (Tempo-anon) [inline src/ui/command-ui-handler.ts:38]: commands-container

File: src/ui/fight-ui-handler.ts [hunk 11]
```diff
@@ -34,48 +35,53 @@ export default class FightUiHandler extends UiHandler {
     const ui = this.getUi();
 
     this.movesContainer = this.scene.add.container(18, -38.7);
+    this.movesContainer.setName("moves");
```
Comment (Tempo-anon) [inline src/ui/fight-ui-handler.ts:38]: I think it's helpful to have the names match the variable names. Places here and on line 42 should have container in the name for easier debugging

File: src/ui/fight-ui-handler.ts [hunk 12]
```diff
@@ -236,10 +243,12 @@ export default class FightUiHandler extends UiHandler {
 
     for (let moveIndex = 0; moveIndex < 4; moveIndex++) {
       const moveText = addTextObject(this.scene, moveIndex % 2 === 0 ? 0 : 100, moveIndex < 2 ? 0 : 16, "-", TextStyle.WINDOW);
+      moveText.setName("-");
```
Comment (Tempo-anon) [inline src/ui/fight-ui-handler.ts:246]: this name hurts

File: src/ui/menu-ui-handler.ts [hunk 13]
```diff
@@ -62,10 +63,16 @@ export default class MenuUiHandler extends MessageUiHandler {
     }
 
     this.menuContainer = this.scene.add.container(1, -(this.scene.game.canvas.height / 6) + 1);
-
+    this.menuContainer.setName("menu");
```
Comment (Tempo-anon) [inline src/ui/menu-ui-handler.ts:66]: menu-container

File: src/ui/menu-ui-handler.ts [hunk 14]
```diff
@@ -62,10 +63,16 @@ export default class MenuUiHandler extends MessageUiHandler {
...
     this.menuContainer.setInteractive(new Phaser.Geom.Rectangle(0, 0, this.scene.game.canvas.width / 6, this.scene.game.canvas.height / 6), Phaser.Geom.Rectangle.Contains);
 
+    this.menuOverlay = new Phaser.GameObjects.Rectangle(this.scene, -1, -1, this.scene.scaledCanvas.width, this.scene.scaledCanvas.height, 0xffffff, 0.3);
+    this.menuOverlay.setName("menu-overlay");
+    this.menuOverlay.setOrigin(0,0);
+    this.menuContainer.add(this.menuOverlay);
+
     const menuMessageText = addTextObject(this.scene, 8, 8, "", TextStyle.WINDOW, { maxLines: 2 });
+    menuMessageText.setName("menu-message");
```
Comment (Tempo-anon) [inline src/ui/menu-ui-handler.ts:75]: menu-message-text

File: src/ui/menu-ui-handler.ts [hunk 15]
```diff
@@ -84,6 +91,7 @@ export default class MenuUiHandler extends MessageUiHandler {
     ui.add(this.menuContainer);
 
     this.menuMessageBoxContainer = this.scene.add.container(0, 130);
+    this.menuMessageBoxContainer.setName("menu-message-box");
```
Comment (Tempo-anon) [inline src/ui/menu-ui-handler.ts:94]: menu-message-box-container

File: src/ui/modifier-select-ui-handler.ts [hunk 16]
```diff
@@ -47,7 +47,7 @@ export default class ModifierSelectUiHandler extends AwaitableUiHandler {
     ui.add(this.modifierContainer);
 
     this.transferButtonContainer = this.scene.add.container((this.scene.game.canvas.width / 6) - 71, -64);
-    this.transferButtonContainer.setName("container-transfer-btn");
+    this.transferButtonContainer.setName("transfer-btn");
```
Comment (Tempo-anon) [inline src/ui/modifier-select-ui-handler.ts:50]: transfer-button-container? I don't think there's a need to abbreviate here and on line 60 and line 70

File: src/ui/party-ui-handler.ts [hunk 17]
```diff
@@ -161,31 +161,37 @@ export default class PartyUiHandler extends MessageUiHandler {
     const ui = this.getUi();
 
     const partyContainer = this.scene.add.container(0, 0);
+    partyContainer.setName("party");
```
Comment (Tempo-anon) [inline src/ui/party-ui-handler.ts:164]: container in the name here and rest of the file

**Review by Tempo-anon — COMMENTED** [review]
Looks fine to me and I sort of gave up after going through 80% of the PR, but I thought names were supposed to match up with the variable names so it'd be easier to find when debugging

File: public/images/pokemon/678msb.json
Comment (brain-frog) [reply public/images/pokemon/678msb.json:1]: can do

File: src/battle-scene.ts (hunk 1 above)
Comment (brain-frog) [reply src/battle-scene.ts:378]: so I removed all the containers because the debug mode also shows the type and it felt repetitive 
for example:
debug

**Comment by brain-frog** [comment]
@Tempo-anon I'm going to resolve all the `setName` related comments since every game object has a [type property](https://newdocs.phaser.io/docs/3.80.0/Phaser.GameObjects.GameObject#type) that says what type it is, no need to reiterate it in the name.

File: src/ui/abstact-option-select-ui-handler.ts (hunk 5 above)
Comment (brain-frog) [reply src/ui/abstact-option-select-ui-handler.ts:87]: when including type I've been having it at the front of the name.

File: src/ui/fight-ui-handler.ts (hunk 12 above)
Comment (brain-frog) [reply src/ui/fight-ui-handler.ts:246]: yeah I'll make this something better lol it was a placeholder

File: src/battle-scene.ts (hunk 2 above)
Comment (brain-frog) [reply src/battle-scene.ts:422]: pb tray is a container

**Review by Tempo-anon — APPROVED** [review]
Discussed offline. Most of the changes in this PR are just adding names

**Pull request merged**